### DISCUSSION TITLE
Feature removal: ambiguous layout file search

### DISCUSSION THREAD

**Discussion by zachleat** [11ty/eleventy#4033] [Enhancement Queue]
> If you omit the file extension (for example layout: mylayout), Eleventy will cycle through all of the supported template formats (mylayout.*) to look for a matching layout file.

https://www.11ty.dev/docs/layouts/#omitting-the-layouts-file-extension

I’d like to phase out the above feature and make it not enabled by default, with the option to re-enable via config option.

This feature requires potentially 10 or 11 file system checks (one per template extension) and the ordering is ambiguous. If you have a base.liquid and base.njk—which does it select? I think for this specific case it’s better to be explicit, especially as we improve incremental builds.

**Comment by zachleat** [comment]
Docs for this at https://www.11ty.dev/docs/layouts/#omitting-the-layouts-file-extension

**Comment by j-f1** [comment]
> this functionality will likely may be removed

Looks like you have one or two extra words there :)

**Comment by zachleat** [comment]
Pushed a fix, thank you @j-f1!

**Comment by zachleat** [comment]
Notably, this is being reverted!

https://github.com/11ty/eleventy/issues/2708

**Comment by zachleat** [comment]
Putting this one back into the enhancement queue

> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!
